### PR TITLE
Continue widget refactoring

### PR DESCRIPTION
**Pull request by alejoe91**
### Notation

Renamed `legacy_mpl_widgets` to `_legacy_mpl_widgets`. Widgets ported to new backend system are renamed with an ending underscore (e.g. `_legacy_mpl_widgets/timeseries_.py`)

### Updates

- Ported to new widgets backends:
    - amplitudes (mpl, ipw, sv)
    - spike locations (mpl, ipw, sv)
    - unit locations (mpl, ipw, sv)
    - quality metrics (mpl, ipw, sv)
    - template metrics (mpl, ipw, sv)
    - similarity (mpl, sv)

- Added histograms to amplitudes
- Added probe layout to templates/waveforms
- Implemented `plot_spikes_on_traces`
- Implemented Unit and Channel controllers for ipywidgets + various options
- Added tests also for sortingview backend using test bucket
- Renamed `AmplitudeWidget` --> `AmplitudesWidget`
- Renamed `UnitTemplateWidget` --> `UnitTemplatesWidget`

### Notes

The ipywidget backend is working well, but tests fail. Needs investigating

### CODE REVIEW

File: spikeinterface/widgets/ipywidgets/utils.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+
```
Comment (samuelgarcia) [inline spikeinterface/widgets/ipywidgets/utils.py:1]: For this module I think I would done classes get value retriever.
But lets go with this approach for now.

**Comment by samuelgarcia** [comment]
We need to check that the `export_report()` because the tests are not trigerred.

**Comment by samuelgarcia** [comment]
Where is the plot_amplitudes_distribution ?

**Comment by alejoe91** [comment]
> Where is the plot_amplitudes_distribution ?

It's incorporated in the plot_amplitudes (optional for mpl and interactive for ipw). 

I'll test with the exporters locally (or we could trigger exporters if widgets is changed)

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**